File: Python/문자열 내 마음대로 정렬하기.py
def solution(strings, n):
    answer = []
    letters = []
    for string in strings:
        letters.append(string[n])
    letters = sorted(list(set(letters)))
    for letter in letters:
        for string in sorted(strings):
            if string[n] == letter:
                answer.append(string)
    return answer

# sorted(array, key) key,
# lambda 매개변수 : 표현식
# def solution(strings, n):
#     return sorted(strings, key=lambda strings:strings[n])

File: Cplusplus/문자열 내 마음대로 정렬하기.cpp
//22.01.17 문자열 내 마음대로 정하기 by CJHofficial
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int index;

bool mysort(string s1, string s2) {
    if (s1[index] < s2[index]) return true;
    else if (s1[index] == s2[index]) {
        return s1 < s2;
    }
    else return false;
}
vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    index = n;
    sort(strings.begin(), strings.end(), mysort);
    return strings;
}